core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -248,11 +248,15 @@ public void onRunFinish(boolean succeeded, BatchSourceContext context) {
       "Example - For tableExpiryPeriod = 30, data before 30 days get deleted from the table.")
     private final String tableExpiryPeriod;
 
+    @Description("Existing hdfs folder path having read and write access to the current User, required for internal" +
+      "computation of the plugin. Default value is /tmp.")
```

Comment:
if it defaults to ``/tmp`` (which I think is correct behavior), this should be a ``@Nullable`` field. So you should not have that precondition on line 296. Instead it should really default to ``/tmp``